### PR TITLE
Adds Direct VPC egress support for Cloud Run V2 resources

### PR DESCRIPTION
**Pull request by yanweiguo**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15568

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudrunv2: added fields `network_interfaces` to resource `google_cloud_run_v2_job` to support Direct VPC egress.
```

```release-note:enhancement
cloudrunv2: added fields `network_interfaces` to resource `google_cloud_run_v2_service` to support Direct VPC egress.
```

### CODE REVIEW

**Review by shuyama1 — COMMENTED** [review]
Are any of these fields support in-place update? If so, would you mind adding/ modifying an update test to the update functionality of it?

File: mmv1/templates/terraform/examples/cloudrunv2_job_directvpc.tf.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  template {
+    template{
+      containers {
+        image = "us-docker.pkg.dev/cloudrun/container/hello"
+      }
+      vpc_access {
+        network_interfaces {
+          network = "default"
+          subnetwork = "default"
+          tags = ["tag1"]
```
[the remark this answers is not in the document]
Comment (yanweiguo) [reply mmv1/templates/terraform/examples/cloudrunv2_job_directvpc.tf.erb]: Done

**Comment by yanweiguo** [comment]
> Are any of these fields support in-place update? If so, would you mind adding/ modifying an update test to the update functionality of it?

Added tests. For Service, I can only use existing "default" network. For Job, non-existing networks are fine.

While adding the tests, found that existing `egress` field lacks default setting. Added `default_from_api: true` since our API set a default value for this field.

**Review by shuyama1 — APPROVED** [review]
Thank you!

**Pull request merged**